server/image_pull.go

```diff
@@ -67,11 +68,20 @@ func (s *Server) PullImage(ctx context.Context, req *pb.PullImageRequest) (*pb.P
 		}
 
 		// let's be smart, docker doesn't repull if image already exists.
-		_, err = s.StorageImageServer().ImageStatus(s.ImageContext(), img)
+		var storedImage *storage.ImageResult
+		storedImage, err = s.StorageImageServer().ImageStatus(s.ImageContext(), img)
 		if err == nil {
-			logrus.Debugf("image %s already in store, skipping pull", img)
-			pulled = img
-			break
+			tmpImg, err := s.StorageImageServer().PrepareImage(s.ImageContext(), img, options)
+			if err != nil {
+				logrus.Debugf("error preparing image, will skip pull: %v", err)
```

Comment:
Should this be at info or error level?  We shouldn't hit this under normal circumstances or when the image already exists right?

Reply:
all this is actually a best-effort, so we should proceed only when `err == nil`

Reply:
Just in case, I was questioning whether we should change logrus.Debugf to logrus.info or logurus.error.  I agree in testing for the error, I just wonder if we should change the logging level.